cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,339 @@
+
+    Focuses specifically on standard AssumeRole events, providing optimized processing
+    for the most common role assumption scenario.
+
+    This function performs the complete transformation pipeline:
+    1. Extract role assumption events from CloudTrail AssumeRole data
+    2. Aggregate events by (source_principal, destination_principal) pairs
+    3. Return aggregated relationships ready for loading
+
+    :type events: List[Dict[str, Any]]
+    :param events: List of raw CloudTrail AssumeRole events from lookup_events API
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: List[Dict[str, Any]]
+    :return: List of aggregated role assumption relationships ready for loading
+    """
+    aggregated: Dict[tuple, Dict[str, Any]] = {}
+    logger.info(
+        f"Transforming {len(events)} CloudTrail AssumeRole events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+
+        # Extract source and destination principals
+        cloudtrail_event = json.loads(event["CloudTrailEvent"])
+
+        # Extract source principal with error handling (UserIdentity may be missing in some events)
+        try:
+            source_principal = cloudtrail_event["userIdentity"]["arn"]
+        except KeyError:
+            logger.debug(
+                f"Skipping CloudTrail AssumeRole event due to missing UserIdentity.arn. Event: {event.get('EventId', 'unknown')}"
+            )
+            continue
+
+        destination_principal = cloudtrail_event["requestParameters"]["roleArn"]
+
+        normalized_source_principal = _convert_assumed_role_arn_to_role_arn(
+            source_principal
+        )
+        normalized_destination_principal = _convert_assumed_role_arn_to_role_arn(
+            destination_principal
+        )
+        event_time = event.get("EventTime")
+
+        key = (normalized_source_principal, normalized_destination_principal)
+        if key in aggregated:
+            aggregated[key]["times_used"] += 1
+            aggregated[key]["assume_role_count"] += 1  # All events are AssumeRole
+            if event_time < aggregated[key]["first_seen_in_time_window"]:
+                aggregated[key]["first_seen_in_time_window"] = event_time
+            if event_time > aggregated[key]["last_used"]:
+                aggregated[key]["last_used"] = event_time
+        else:
+            aggregated[key] = {
+                "source_principal_arn": normalized_source_principal,
+                "destination_principal_arn": normalized_destination_principal,
+                "times_used": 1,
+                "first_seen_in_time_window": event_time,
+                "last_used": event_time,
+                "event_types": ["AssumeRole"],
+                "assume_role_count": 1,
+                "saml_count": 0,
+                "web_identity_count": 0,
```

Comment:
[block]  Remove these fields, they're not used  because this sync function is only for assumerole